File: task/src/main/java/com/goexp/galgame/data/task/handler/game/LocalGameHandler.java
package com.goexp.galgame.data.task.handler.game;

import com.goexp.galgame.data.Config;
import com.goexp.galgame.data.piplline.core.Message;
import com.goexp.galgame.data.piplline.core.MessageQueueProxy;
import com.goexp.galgame.data.piplline.handler.DefaultMessageHandler;
import com.goexp.galgame.data.task.handler.MesType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;
import java.nio.file.Files;
import java.util.Map;
import java.util.Objects;

public class LocalGameHandler extends DefaultMessageHandler<Integer> {

    final private Logger logger = LoggerFactory.getLogger(LocalGameHandler.class);


    @Override
    public void process(final Message<Integer> message, MessageQueueProxy<Message> msgQueue) {

        var gid = message.entity;
        logger.debug("<Game> {}", gid);

        var bytes = Map.entry(gid, Objects.requireNonNull(getContent(gid), gid.toString()));

        msgQueue.offer(new Message<>(MesType.ContentBytes, bytes));
    }

    private byte[] getContent(int id) {
        try {
            var path = Config.GAME_CACHE_ROOT.resolve(String.format("%d.bytes", id));

            return Files.readAllBytes(path);

        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }


}

File: task/src/main/scala/com/goexp/galgame/data/task/handler/game/LocalGameHandler.scala
package com.goexp.galgame.data.task.handler.game

import java.io.IOException
import java.nio.file.Files
import java.util.Objects

import com.goexp.galgame.data.Config
import com.goexp.galgame.data.piplline.core.{Message, MessageQueueProxy}
import com.goexp.galgame.data.piplline.handler.DefaultMessageHandler
import com.goexp.galgame.data.task.handler.MesType
import org.slf4j.LoggerFactory

class LocalGameHandler extends DefaultMessageHandler[Int] {
  final private val logger = LoggerFactory.getLogger(classOf[LocalGameHandler])

  override def process(message: Message[Int], msgQueue: MessageQueueProxy[Message[_]]) = {
    val gid = message.entity
    logger.debug("<Game> {}", gid)

    val bytes = (gid, Objects.requireNonNull(getContent(gid), gid.toString))
    msgQueue.offer(new Message[(Int, Array[Byte])](MesType.ContentBytes, bytes))
  }

  private def getContent(id: Int) =
    try {
      val path = Config.GAME_CACHE_ROOT.resolve(s"$id.bytes")
      Files.readAllBytes(path)
    } catch {
      case e: IOException =>
        e.printStackTrace()
        throw new RuntimeException(e)
    }
}